sql/src/test/java/io/crate/planner/PlanPrinterTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner;
+
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+public class PlanPrinterTest extends CrateDummyClusterServiceUnitTest {
+
+    private SQLExecutor e;
+
+    @Before
+    public void setUpExecutor() {
+        e = SQLExecutor.builder(clusterService).enableDefaultTables().build();
+    }
+
+    private Map<String, Object> printPlan(String stmt) {
+        return PlanPrinter.objectMap(e.plan(stmt));
+    }
+
+    @Test
+    public void testQueryTheFetch() {
+        Map<String, Object> map = printPlan("select * " +
+                                            "from t1 " +
+                                            "where t1.x > 10 " +
+                                            "order by a " +
+                                            "limit 10");
+        String mapStr = map.toString();
+        assertThat(mapStr, containsString(
+            "{QueryThenFetch={type=executionPlan, subPlan={Collect={type=executionPlan, " +
+               "collectPhase={COLLECT={type=executionPhase, id=0, executionNodes=[n1], " +
+                   "distribution={distributedByColumn=0, type=BROADCAST}, toCollect=Ref{doc.t1._fetchid, long}, Ref{doc.t1.a, string}, " +
```

Comment:
I wonder if it would make more sense to just test the "shape" (that right properties are there)

Otherwise if we change the string representations this will be annoying to maintain.

Reply:
@mfussenegger Testing the whole String is not an elegant and flexible solution.
On the other hand, at first, I started to test the Object tree within the map. The result was also quite ugly and quite troublesome also to maintain if something is changing in the logical/execution plan.

I chose to keep the assertions on the string representation because they seem, IMHO, more readable than many explicit assertions on every object of the tree.